parse_nucleotide_fasta.c

```diff
@@ -0,0 +1,263 @@
+            for (i = 0; i < 5; i++) {
+                site_values[i] = 0;
+            }
+            chunk_num++;
+        }
+        char site = read[site_index];
+        if (site == 'A' || site == 'a') {
+            site_values[0]++;
+            site_values[4]++;
+        } else if (site == 'T' || site == 't') {
+            site_values[1]++;
+            site_values[4]++;
+        } else if (site == 'C' || site == 'c') {
+            site_values[2]++;
+            site_values[4]++;
+        } else if (site == 'G' || site == 'g') {
+            site_values[3]++;
+            site_values[4]++;
+        } else {
+          // Do Nothing
+        }
+        site_index++;
+    }
+
+    return chunk_num;
+}
+
+int find_all_kmer_permutations() {
+    // Finds all the permutations for all K-mers of size K
+}
+
+/******************************
+*   Searches recursivly the bases in two reads (the current read that the program is on and the prevous read).
+*   It will output 1 if the search gets to the first index in the word without matching errors (meaning the
+*       read and the word match) and 0 otherwise.
+*******************************/
+int recursive_char_search(struct Flags* flags, int word_index, char* curr_read, char* prev_read, int index) {
+    if (word_index < 0) {
+        return 1;
+    }
+    if (index < 0) {
+        int prev_read_size = strlen(prev_read);
+        if (prev_read[prev_read_size + index] == flags[3].flag_value[word_index]) {
+            return recursive_char_search(flags, word_index - 1, curr_read, prev_read, index - 1);
+        } else {
+            return 0;
+        }
+    } else {
+        if (curr_read[index] == flags[3].flag_value[word_index]) {
+            return recursive_char_search(flags, word_index - 1, curr_read, prev_read, index - 1);
+        } else {
+            return 0;
+        }
+    }
+}
+
+/******************************
+*   Counts the number of times a word is found in the fasta file
+*******************************/
+int matching_occurences_count(struct Flags* flags, char* curr_read, char* prev_read) {
+    int occurence_count = 0;
+    int curr_read_size = strlen(curr_read);
+    int word_size = strlen(flags[3].flag_value);
+    int i;
+    if (strcmp(prev_read, "")) {
+        for (i = 0; i < curr_read_size; i++) {
+            occurence_count += recursive_char_search(flags, word_size - 1, curr_read, prev_read, i);
+        }
+    } else {
+        for (i = word_size - 1; i < curr_read_size; i++) {
+            occurence_count += recursive_char_search(flags, word_size - 1, curr_read, prev_read, i);
+        }
+    }
+
+    return occurence_count;
+}
+
+/******************************
+*   Reads through fasta file and runs other functions depending on the input flags passed in.
+*       This function opens the file and reads through each line, It will pass the lines to other functions which do stuff
+*******************************/
+int parse_fasta(struct Flags* flags) {
```

Comment:
Why does parse_fasta return an integer? It's never used when you call it down in `main()`, plus you always return 0